File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int count = 0;
        int num = 0;
        count = 1;
        num = nums[0];
        for (int i=1; i < nums.length; i++) {
            if (nums[i] == num) {
                count++;
            } else {
                count--;
            }
            if (count == 0) {
                count = 1;
                num = nums[i];
            }
        }
        return num;
    }
}

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 1
        majority_elem = nums[0]
        for i in range(1, len(nums)):
            num = nums[i]
            if num == majority_elem:
                count += 1
            else:
                if count == 1:
                    count = 1
                    majority_elem = num
                else:
                    count -= 1
        return majority_elem